zun/scheduler/filters/blazar_filter.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2017 OpenStack Foundation
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import re
+
+from oslo_log.log import logging
+
+from zun.common import context as zun_context
+from zun.scheduler import filters
+from zun.scheduler.client import report
+
+LOG = logging.getLogger(__name__)
+
+BLAZAR_PLACEMENT_TRAIT_PATTERN = "^CUSTOM_RESERVATION_([A-Z0-9_]*)_PROJECT_([A-Z0-9_]*)$"
```

Comment:
You can use named capture groups like this:

```suggestion
BLAZAR_PLACEMENT_TRAIT_PATTERN = "^CUSTOM_RESERVATION_(?P<reservation_id>[A-Z0-9_]*)_PROJECT_(?P<project_id>[A-Z0-9_]*)$"
```

which should then allow you to do like `match.group('reservation_id')`. Not really a suggestion, just a note in case it is handy in the future.